build-system-tests/package.json

```diff
@@ -3,7 +3,32 @@
   "version": "1.0.0",
   "private": true,
   "devDependencies": {
+    "@supercharge/promise-pool": "^2.4.0",
+    "commander": "^10.0.1",
     "json": "^11.0.0",
     "strip-json-comments-cli": "^2.0.2"
+  },
+  "scripts": {
+    "setup:react": "./scripts/setup-mega-app.sh",
+    "setup:angular": "./scripts/setup-mega-app.sh -F angular",
+    "setup:vue": "./scripts/setup-mega-app.sh -F vue",
+    "setup:react-native": "./scripts/setup-mega-app.sh -F react-native",
+    "react-latest-cra-latest-ts": "npm run setup:react",
+    "react-latest-next-latest-ts": "npm run setup:react -- -B next",
+    "react-latest-vite-latest-ts": "npm run setup:react -- -B vite",
+    "react-16-cra-latest-ts": "npm run setup:react -- -f 16",
+    "react-latest-cra-latest-js": "npm run setup:react -- -l js",
+    "react-17-next-11-ts": "npm run setup:react -- -f 17 -B next -b 11",
+    "react-latest-vite-2-ts": "npm run setup:react -- -B vite -b 2",
```

Comment:
Since these are utility scripts for our convenience, I think we should focus on local development DX than trying to incorporate `build-system-tests` workflow details bleed into this.

```
"setup:react:cra": "npm run setup:react -- -B cra",
"setup:react:vite": "npm run setup:react -- -B vite",
"setup:react:next": "npm run setup:react -- -B next",

"setup:vue:vite": "npm run setup:vue -- -B vite",
"setup:vue:nuxt": "npm run setup:vue -- -B nuxt",

"setup:angular:cli": "npm run setup:angular -- -B angular-cli", 
```

I think this easier to read for local usages, and shows how they're structured (`<framework>:<build-tool>`). Developers then can easily append flags:

```
npm run setup:react:next -- --framework-version 17 --build-tool-version 11
```

which feels more flexible to me than predefined scripts for each configuration.


For the GHA workflow, it can directly use all the flags since convenience isn't much of concern:

```
run: npm run setup -- -F ${{ matrix.framework }} -f ${{ matrix.framework-version }} ...rest of the flags
``` 

I think this is more acceptable now that we have all-in-one script.

Reply:
It does look better and I can add those scripts (setup:react:cra etc.) there, but it'd be also helpful to keep the pre-configed MegaApp names there,  like `react-latest-vite-2-ts`, so that if there's any issue with any app, the oncall engineer can set the MegaApp without thinking.

What do you think?

Reply:
That's fair, it's a deal 😈

Reply:
Can you add a README.md here on how to use these scripts?

Reply:
Done 42694de83b74061558e8169cccdaecb01636fbd2